src/passes/OptimizeInstructions.cpp

```diff
@@ -1227,6 +1224,40 @@ struct OptimizeInstructions
           right->value = Literal::makeSingleZero(type);
           return right;
         }
+        // bool(x) | 1  ==>  1
+        // bool(x) & 1  ==>  bool(x)
+        // bool(x) == 1  ==>  bool(x)
+        // bool(x) != 1  ==>  !bool(x)
+        if (Bits::getMaxBits(binary->left, this) == 1) {
+          switch (binary->op) {
+            case OrInt32:
+            case OrInt64: {
+              // bool(x) | 1  ==>  1
+              return binary->right;
```

Comment:
Probably we should check side effects for `binary->left` here, wdyt?

Reply:
Ah right, yes checking for side effects is important.